Aicha Sidiya - S20106146

Layan Jaman - S20106590

##Question A:

In [ ]:
!pip install -U assemblyai

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.3/63.3 kB 1.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.0/75.0 kB 5.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 130.2/130.2 kB 15.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 76.9/76.9 kB 9.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.3/58.3 kB 8.3 MB/s eta 0:00:00


In [ ]:
import assemblyai as aai

aai.settings.api_key = "f59718b9936e433286f32f57710b734e"
transcriber = aai.Transcriber()

transcript = transcriber.transcribe("https://storage.googleapis.com/aai-web-samples/news.mp4")
# transcript = transcriber.transcribe("./my-local-audio-file.wav")

print(transcript.text)

I'm David Curley at the Smithsonian Aaron Space Museum, where we are marking 50 years since man landed and walked on the moon in a lander just like this one. We are going to show you some of the actual ABC News coverage from 50 years ago during that eight day mission of this remarkable achievement. Apollo Eleven's lander, the Eagle, would be the first manned craft to land on the moon. For training, NASA came with an unusual contraption. Neil Armstrong actually had to eject from it once, and then he had a couple of successful flights. ABC News Anchor at the time, 50 years ago, frank Reynolds with a look at that unusual trainer. Apollo Eleven commander Neil Armstrong is at the controls of a lunar landing training vehicle. Testing the reaction control jets, these thrusters stabilize the Lem during landing and takeoff. The LLTV is designed to simulate the behavior of the Lem as it lands in the moon's gravity. Lunar gravity is one 6th that of the Earth's. Neil Armstrong flew one of these ve

In [ ]:
config = aai.TranscriptionConfig(
  summarization=True,
  summary_model=aai.SummarizationModel.informative, # optional
  summary_type=aai.SummarizationType.bullets # optional
)

transcriber = aai.Transcriber(config=config)

In [ ]:
transcript = transcriber.transcribe("/content/Metaverse.mp3")

In [ ]:
print(transcript.text)

The Metaverse, a digital world where anything is possible. Imagine a place where you can interact with others, play games, and even work, all in a virtual setting. The metaverse is more than just a digital world. It's a revolution. It's changing the way we think about communication, entertainment, and even commerce. With the Metaverse, the possibilities are endless. It's a new frontier, and we're just beginning to scratch the surface of what it can do. Join the Metaverse revolution today and see what the future holds. The Metaverse, where the future is now.


In [ ]:
text = transcript.text

In [ ]:
print(transcript.summary)

- The Metaverse is more than just a digital world. It's changing the way we think about communication, entertainment, and even commerce. Join the Metaverse revolution today and see what the future holds.


In [ ]:
summary = transcript.summary

In [ ]:
print("Transcribed text length: ", len(text) , "\nSummarized text length: ", len(summary), "\nDifference: ", len(text)-len(summary))

Transcribed text length:  563 
Summarized text length:  203 
Difference:  360


In this task we used assembly ai to transcribe and an audio then we summarized it after summarization we can see that the length became 203, it summurized about 360 tokens.

Topic Detection

In [ ]:
import assemblyai as aai

audio_url = "https://github.com/AssemblyAI-Examples/audio-examples/raw/main/20230607_me_canadian_wildfires.mp3"

config = aai.TranscriptionConfig(iab_categories=True)

transcript = aai.Transcriber().transcribe(audio_url, config)

# Get the parts of the transcript that were tagged with topics
for result in transcript.iab_categories.results:
    print(result.text)
    print(f"Timestamp: {result.timestamp.start} - {result.timestamp.end}")
    for label in result.labels:
        print(f"{label.label} ({label.relevance})")

# Get a summary of all topics in the transcript
for topic, relevance in transcript.iab_categories.summary.items():
    print(f"Audio is {relevance * 100}% relevant to {topic}")

Smoke from hundreds of wildfires in Canada is triggering air quality alerts throughout the US. Skylines from Maine to Maryland to Minnesota are gray and smoggy. And in some places, the air quality warnings include the warning to stay inside. We wanted to better understand what's happening here and why, so we called Peter DiCarlo, an associate professor in the Department of Environmental Health and Engineering at Johns Hopkins University. Good morning, professor. Good morning.
Timestamp: 250 - 28920
Home&Garden>IndoorEnvironmentalQuality (0.9881861209869385)
NewsAndPolitics>Weather (0.5561717748641968)
MedicalHealth>DiseasesAndConditions>LungAndRespiratoryHealth (0.00426860898733139)
NewsAndPolitics>Disasters (0.0034602025989443064)
BusinessAndFinance>Business>GreenSolutions (0.0025904506910592318)
MedicalHealth>DiseasesAndConditions>Cancer (0.0015328084118664265)
Science>Environment (0.001438859268091619)
Travel>TravelLocations>PolarTravel (0.0013438508613035083)
MedicalHealth>Diseases

In [ ]:
import assemblyai as aai

audio_url = "/content/Metaverse.mp3"

config = aai.TranscriptionConfig(iab_categories=True)

transcript = aai.Transcriber().transcribe(audio_url, config)

# Get the parts of the transcript that were tagged with topics
for result in transcript.iab_categories.results:
    print(result.text)
    print(f"Timestamp: {result.timestamp.start} - {result.timestamp.end}")
    for label in result.labels:
        print(f"{label.label} ({label.relevance})")

# Get a summary of all topics in the transcript
for topic, relevance in transcript.iab_categories.summary.items():
    print(f"Audio is {relevance * 100}% relevant to {topic}")

The Metaverse, a digital world where anything is possible. Imagine a place where you can interact with others, play games, and even work, all in a virtual setting. The metaverse is more than just a digital world. It's a revolution. It's changing the way we think about communication, entertainment, and even commerce. With the Metaverse, the possibilities are endless. It's a new frontier, and we're just beginning to scratch the surface of what it can do.
Timestamp: 970 - 24902
VideoGaming>VideoGameGenres>MMOs (0.8674062490463257)
Hobbies&Interests>ContentProduction (0.8379799723625183)
Technology&Computing>Computing>ComputerSoftwareAndApplications>WebConferencing (0.4135737121105194)
Technology&Computing>Computing>Internet>SocialNetworking (0.2242642045021057)
Technology&Computing>Computing>Internet>ITAndInternetSupport (0.1525728702545166)
Technology&Computing>Computing>Internet (0.08376971632242203)
Careers>RemoteWorking (0.07496453076601028)
Technology&Computing>Computing>Internet>Int

Using assembly ai we were able to identify the main topics in our vedio on metaverse. The video discussed mostly topics related to video games, hobbies, and technology.

##Question B:

In [ ]:
from transformers import GPT2Tokenizer, GPT2LMHeadModel, pipeline

def generate_text_with_gpt2(prompt, max_length=100, summarization_max_length=50):
    # Load the pre-trained GPT-2 model and tokenizer
    model_name = "gpt2"
    tokenizer = GPT2Tokenizer.from_pretrained(model_name)
    model = GPT2LMHeadModel.from_pretrained(model_name)

    # Encode the prompt and generate text using GPT-2
    input_ids = tokenizer.encode(prompt, return_tensors="pt")
    output_ids = model.generate(input_ids, max_length=max_length, num_return_sequences=1)
    output_text = tokenizer.decode(output_ids[0], skip_special_tokens=True)

    # Use summarization pipeline to summarize the generated text
    summarizer = pipeline("summarization")
    summarized_text = summarizer(output_text, max_length=summarization_max_length, min_length=5, length_penalty=2.0, num_beams=4)[0]['summary_text']

    return summarized_text

# Input paragraph
paragraph = """
Artificial intelligence (AI) is revolutionizing the way we live and work. It involves the development of computer systems that can perform tasks that typically require human intelligence. From natural language processing to image recognition, AI applications are diverse and impactful.

One of the key challenges in AI is ensuring ethical and responsible use. As AI systems become more sophisticated, there is a growing need for guidelines and regulations to govern their deployment. Privacy concerns, bias in algorithms, and the potential for job displacement are important considerations in the ethical use of AI.

Despite challenges, AI holds great promise for solving complex problems and improving efficiency across various industries. From healthcare to finance, AI applications are helping make better decisions and streamline processes. As the field continues to advance, it is crucial to balance innovation with ethical considerations to ensure a positive and inclusive impact on society.
"""

# Generate and summarize text
generated_text = generate_text_with_gpt2(paragraph)

print("Input Paragraph:")
print(paragraph)
print("\nGenerated and Summarized Text:")
print(generated_text)


vocab.json:   0%|          | 0.00/1.04M [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

config.json:   0%|          | 0.00/665 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/548M [00:00<?, ?B/s]

generation_config.json:   0%|          | 0.00/124 [00:00<?, ?B/s]

The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
/usr/local/lib/python3.10/dist-packages/transformers/generation/utils.py:1281: UserWarning: Input length of input_ids is 174, but `max_length` is set to 100. This can lead to unexpected behavior. You should consider increasing `max_new_tokens`.
  warnings.warn(
No model was supplied, defaulted to sshleifer/distilbart-cnn-12-6 and revision a4f8f3e (https://huggingface.co/sshleifer/distilbart-cnn-12-6).
Using a pipeline without specifying a model name and revision in production is not recommended.


config.json:   0%|          | 0.00/1.80k [00:00<?, ?B/s]

pytorch_model.bin:   0%|          | 0.00/1.22G [00:00<?, ?B/s]

In this task we used the GPT-2 model to summarize a paragraph that is given from the user. It will generate a summarized text based on an input paragraph. It then employs a summarization model to provide a concise summary of the generated text. The goal is to showcase how GPT-2 can be used for both text generation and summarization, particularly when given a paragraph as input.

Part C

In this quiz we used 2 apis to summarize text, the first was assembly ai and the second is gpt, the benifit of assembly ai is that it can perform the summarization on audio by transcribing them to text while gpt use only text based data.